test/fixtures/test-runner/output/default_output.js

```diff
@@ -9,7 +9,9 @@ const test = require('node:test');
 test('should pass', () => {});
 test('should fail', () => { throw new Error('fail'); });
 test('should skip', { skip: true }, () => {});
-test('parent', () => {
-  test('should fail', () => { throw new Error('fail'); });
-  test('should pass but parent fail', () => {});
+test('parent', (t) => {
+  t.test('should fail', () => { throw new Error('fail'); });
+  t.test('should pass but parent fail', (t, done) => {
```

Comment:
Shouldn't this work even when using `test` directly?

Reply:
This test is currently doing two things wrong:
- It's using `test()` inside of `test()` instead of `t.test()`.
- It's not awaiting things inside of the `test()` call.

Prior to this change, it "works" in the sense that the snapshotted output of the `should pass but parent fail` subtest is a `test did not finish before its parent and was cancelled` error. If you run the same code with `t.test()` (which does not have this extra `await`) that subtest passes.

With the change in this PR, `test()` passes like `t.test()` does. I figured if I needed to update either the test or the snapshot because the test is wrong, I would make the test less wrong.